packages/dev/core/src/Audio/audioEngine.ts

```diff
@@ -175,6 +175,23 @@ export class AudioEngine implements IAudioEngine {
         }
     }
 
+    /** @internal */
+    public _resumeAudioContextOnStateChange(): void {
+        this._audioContext?.addEventListener(

```

Comment:
Does `_audioContext` outlive `AudioEngine`? Is it a problem that we never remove this event listener? Will we have a leak? Also if you enter XR multiple times, will we end up registering the event handler multiple times?

Reply:
The options object passed in prevents this. The `once: true` and 3 second abort signal make sure it's removed.